synapse/cores/common.py

```diff
@@ -457,12 +457,23 @@ def _addDataModels(self, modtups):
 
             for fnam, fnfo, props in modl.get('forms', ()):
 
-                self.addRuntNode('syn:form', fnam, **fnfo)
-                self.addRuntNode('syn:prop', fnam, **fnfo)
+                dmpnfo = self.getPropDef(fnam)[1]
+                dmfnfo = dmpnfo.copy()
+                [dmfnfo.pop(key) for key, valu in list(dmpnfo.items()) if valu is None]
```

Comment:
not sure i see why we're doing this whole copy/pop construct here...  runt nodes are designed to be able to contain non-model props ( that are just not indexed ).  Am i missing something?

Reply:
The default values for props in the model includes some None values which allow for some introspection (in the getPropInfo).  Popping these values out means we don't try to norm a None value, causing a badtypevalu to be thrown.